test/codegen/benchmark_scan_test.cpp

```diff
@@ -0,0 +1,225 @@
+    fprintf(stderr,
+            "Initialization time: %.2f ms, execution time: %.2f ms, Tear down "
+            "time: %.2f ms\n",
+            runtime_stats.init_ms, runtime_stats.plan_ms,
+            runtime_stats.tear_down_ms);
+    fprintf(stderr,
+            "Tuple result size: %d\n",
+            tuple_result_size);
+  }
+};
+
+class BenchmarkScanTest : public PelotonCodeGenTest {
+ public:
+  BenchmarkScanTest() : PelotonCodeGenTest(), num_rows_to_insert(64) {
+    // Load test table
+    LoadTestTable(TestTableId(), num_rows_to_insert);
+  }
+
+  uint32_t NumRowsInTestTable() const { return num_rows_to_insert; }
+
+  uint32_t TestTableId() { return test_table1_id; }
+
+  Stats RunCompiledExperiment(uint32_t num_runs = 5) {
+      // Keep one copy of compile and runtime stats
+      Stats stats;
+
+      for (uint32_t i = 0; i < num_runs; i++) {
+        planner::SeqScanPlan scan{&GetTestTable(TestTableId()), nullptr, {0, 1, 2}};
+
+        // Do binding
+        planner::BindingContext context;
+        scan.PerformBinding(context);
+
+        // We collect the results of the query into an in-memory buffer
+        codegen::BufferingConsumer buffer{{0, 1, 2}, context};
+
+        // COMPILE and execute
+        codegen::QueryStatement::RuntimeStats runtime_stats;
+        codegen::QueryCompiler::CompileStats compile_stats = CompileAndExecute(scan, buffer, reinterpret_cast<char*>(buffer.GetState()), &runtime_stats);
+
+        stats.Merge(compile_stats, runtime_stats,
+            buffer.GetOutputTuples().size());
+      }
+
+      stats.Finalize();
+      return stats;
+    }
+
+    Stats RunInterpretedExperiment(uint32_t num_runs = 5) {
+      // Keep one copy of compile and runtime stats
+      Stats stats;
+
+      for (uint32_t i = 0; i < num_runs; i++) {
+        planner::SeqScanPlan scan{&GetTestTable(TestTableId()), nullptr, {0, 1, 2}};
+        std::vector<std::vector<type::Value>> vals;
+
+        codegen::QueryCompiler::CompileStats compile_stats;
+        codegen::QueryStatement::RuntimeStats runtime_stats;
+
+        auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+        auto txn = txn_manager.BeginTransaction();
+
+        executor::ExecutorContext ctx{txn};
+        executor::SeqScanExecutor executor{&scan, &ctx};
+
+        executor.Init();
+
+        Timer<std::ratio<1, 1000>> timer;
+        timer.Start();
+        while (executor.Execute()) {
+          auto tile = executor.GetOutput();
+          for (oid_t tuple_id : *tile) {
+            const expression::ContainerTuple<executor::LogicalTile> tuple{
+                tile, tuple_id};
+            std::vector<type::Value> tv;
+            for (oid_t col_id : scan.GetColumnIds()) {
+              tv.push_back(tuple.GetValue(col_id));
+            }
+            vals.push_back(std::move(tv));
+          }
+        }
+        timer.Stop();
+        runtime_stats.plan_ms = timer.GetDuration();
```

Comment:
The timer is set before and after the execution which measures the overall time duration. It's a little misleading to store time in runtime_stats.plan_ms. Also, since the compile_stats is not available for interpreted method, compile_stats is not needed to be created within the iteration.